src/pages/index.js

```diff
@@ -0,0 +1,393 @@
+
+// Preview Modal elements
+const previewModal = document.querySelector('#preview-modal');
+const previewModalImageEl = previewModal.querySelector('.modal__image');
+const previewModalCaptionEl = previewModal.querySelector('.modal__caption');
+const previewModalCloseBtn = previewModal.querySelector('.modal__close-btn');
+
+//Card related elements
+const cardTemplate = document.querySelector('#card-template');
+const cardsList = document.querySelector('.cards__list');
+
+// select the avatar modal and its elements
+const avatarModal = document.querySelector('#avatar-modal');
+const avatarForm = avatarModal.querySelector('.modal__form');
+const avatarSubmitBtn = avatarModal.querySelector('.modal__submit-btn');
+const avatarModalCloseBtn = avatarModal.querySelector('.modal__close-btn');
+const avatarInput = avatarModal.querySelector('#profile-avatar-input');
+
+// Delete Modal elements
+const deleteModal = document.querySelector('#delete-modal');
+const deleteForm = deleteModal.querySelector('.modal__form');
+const deleteModalCancelBtn = deleteModal.querySelector('#delete-cancel-btn');
+let selectedCard;
+let selectedCardId;
+
+function openModal(modal) {
+  modal.classList.add('modal_opened');
+  document.addEventListener('keydown', handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove('modal_opened');
+  document.removeEventListener('keydown', handleEscapeKey);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  // editModalSubmitBtn.textContent = 'Saving...';
+  // change text content to "Saving..."
+  const editModalSubmitBtn = evt.submitter;
+  editModalSubmitBtn.textContent = 'Saving...';
+
+  setButtonText(editModalSubmitBtn, true, "Save", "Saving...");
+  // disable the button
+  api.editUserInfo({ name: editModalNameInput.value, about: editModalDescriptionInput.value })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+
+      // todo - call setButtonText instead
+      editModalSubmitBtn.textContent = "Save";
+    });
+}
+
+// todo - implement loading text for all other form submissions
+
+function handleAddCardFormSubmit(evt) {
+  evt.preventDefault();
+
+  const addCardModalSubmitBtn = evt.submitter;
+  addCardModalSubmitBtn.textContent = 'Saving...';
+  setButtonText(addCardModalSubmitBtn, true, "Save", "Saving...");
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      console.log(data);
+
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+    }).catch(console.error)
+    .finally(() => {
+
+      addCardModalSubmitBtn.textContent = "Save";
+    });
+  disableButton(cardSubmitBtn, settings);
+  closeModal(cardModal);
```

Comment:
```js

  disableButton(cardSubmitBtn, settings);
  closeModal(cardModal);
```

These 2 lines should be in `.then` in line 178


* You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**
* You need to **disable** the button only in a `.then` block after a **successful response** to be able to click it again if there is a server error.